apps/parsley/src/components/LogRow/RowRenderer/index.tsx

```diff
@@ -90,13 +90,22 @@ const ParsleyRow: RowRendererFunction = ({ processedLogLines }) => {
     }
 
     if (isSubsectionHeaderRow(processedLogLine)) {
+      let status;
+      // Only show status icon for top-level commands
+      if (processedLogLine.isTopLevelCommand) {
+        status = includesLineNumber(processedLogLine, failingLine)
+          ? SectionStatus.Fail
+          : SectionStatus.Pass;
+      }
```

Comment:
We can do this earlier on achieve the refactor you're mentioning but I don't think it will necessarily be cleaner at this point in time. Currently, the status calculation only happens in this file and is only important for render purposes. 
To get the value to propagate through, `useSections` would have to accept a separate `failingLineNumber` prop, and `FunctionEntry`, `CommandEntry`, `SubsectionHeader` and `SectionHeader` would all need to push along the value.

Reply:
@minnakt I think we should make this ticket when we need the SectionStatus to do operations on ProcessedLogLines. I don't think the refactor will have a great benefit until then since `includesLineNumber` is a constant time operation.

Reply:
it's less about the performance benefits (I assume there are none) and more that I think it makes sense to have all of this information in one place. I can make a ticket since I feel pretty strongly about it